metrics/resource_view.go

```diff
@@ -237,15 +238,19 @@ func optionForResource(r *resource.Resource) (stats.Options, error) {
 func resourceAsString(r *resource.Resource) string {
 	var s strings.Builder
 	l := len(r.Type)
+	keys := make([]string, 0, len(r.Labels))
 	for k, v := range r.Labels {
 		l += len(k) + len(v) + 2
+		keys = append(keys, k)
```

Comment:
Why not move the `fmt.Fprintf` up here, and then you could replace the second loop with `strings.Join(labels, "")`